pkg/arvo/app/chat-hook.hoon

```diff
@@ -33,42 +34,62 @@
       invite-created=_|
   ==
 ::
++$  state-two
+  $:  %2
+      synced=(map path ship)
+      boned=(map wire (list bone))
+      invite-created=_|
+      allow-history=(map path ?)
+  ==
+::
 +$  poke
   $%  [%chat-action chat-action]
       [%permission-action permission-action]
       [%invite-action invite-action]
       [%chat-view-action chat-view-action]
   ==
-::
 --
 ::
-|_  [bol=bowl:gall state-one]
+|_  [bol=bowl:gall state-two]
 ::
 ++  this  .
 ::
 ++  prep
-  |=  old=(unit state-both)
-  ^-  (quip move _this)
+  |=  old=(unit versioned-state)
+  |^  ^-  (quip move _this)
```

Comment:
What's this about?

Reply:
It scopes the migrate state function within this core to show that it's only used here.

Reply:
Oh, durr, I read this as a `^|` somehow. Nevermind!